src/bindings/python/flux/cli/base.py

```diff
@@ -751,7 +752,10 @@ def __init__(self, prog, usage=None, description=None, exclude_io=False):
         self.exitcode = 0
         self.progress = None
         self.watcher = None
+        self.plugins = CLIPluginRegistry().load_plugins(prog)
         self.parser = self.create_parser(prog, usage, description, exclude_io)
+        self.group = self.parser.add_argument_group("Options provided by plugins")
+        self.plugins.add_plugin_option(self.group)
```

Comment:
Adding the `-P, --plugin` option should be pulled out of the function and moved to where all the other options are added.